async.go

```diff
@@ -0,0 +1,33 @@
+package cli
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	stream "github.com/GetStream/stream-chat-go/v5"
+)
+
+func WaitForAsyncCompletion(ctx context.Context, c *stream.Client, taskId string) error {
```

Comment:
I would suggest to have this as a separate command. Channel deletion command (and any other async coming tasks) would return the task ID and we can watch it with something like
```shell
$> stream-cli watch ID
```
which print if the task is still running or is done.

Reply:
honestly, I would challenge that approach.

I agree that we can create a `stream-cli watch ID` command, but I think it's just too annoying to have to run a separate command to check if you could delete a channel. I think it's just a nice touch that it's already built into it.

Reply:
> honestly, I would challenge that approach.

It's appreciated!


What I don't like with the current approach is that:
- if we exceed 10secs we return an error, we should not
- if it takes time to delete the channel for any reasons, i am blocked until it's done (if i don't CTRL+C)

I suggest to take both approaches then:
- don't return an error if it takes more time, make this configurable
- print the task ID
- add a watch command to watch the task
- make this clear in the ouput or documentation that you can terminate the delete command if you want and you will still be able to watch if the deletion is done or not

Reply:
✅